packages/wdio-devtools-service/src/index.js

```diff
@@ -29,23 +36,133 @@ export default class DevToolsService {
+        global.browser.addCommand('disablePerformanceAudits', ::this._disablePerformanceAudits)
+
         /**
-         * enable network and page domain for resource analysis
+         * allow user to work with Puppeteer directly
          */
-        await this.commandHandler.cdp('Network', 'enable')
-        await this.commandHandler.cdp('Page', 'enable')
+        global.browser.addCommand('getPuppeteer',
+            () => this.devtoolsDriver)
+    }
+
+    async beforeCommand (commandName, params) {
+        if (!this.shouldRunPerformanceAudits || !this.traceGatherer || this.traceGatherer.isTracing || !TRACE_COMMANDS.includes(commandName)) {
+            return
+        }
+
+        /**
+         * set browser profile
+         */
+        this._setThrottlingProfile(this.networkThrottling, this.cpuThrottling, this.cacheEnabled)
+
+        const url = commandName === 'navigateTo' ? params[0] : 'click transition'
+        return this.traceGatherer.startTracing(url)
+    }
+
+    async afterCommand (commandName) {
+        if (!this.traceGatherer || !this.traceGatherer.isTracing || !TRACE_COMMANDS.includes(commandName)) {
+            return
+        }
+
+        /**
+         * update custom commands once tracing finishes
+         */
+        this.traceGatherer.once('tracingComplete', (traceEvents) => {
+            const auditor = new Auditor(traceEvents, this.devtoolsGatherer.getLogs())
+            auditor.updateCommands(global.browser)
+        })
+
+        return new Promise((resolve) => {
+            log.info(`Wait until tracing for command ${commandName} finishes`)
+
+            /**
+             * wait until tracing stops
+             */
+            this.traceGatherer.once('tracingFinished', async () => {
+                log.info('Disable throttling')
+                await this._setThrottlingProfile('online', 0, true)
+
+                log.info('continuing with next WebDriver command')
+                resolve()
+            })
+        })
+    }
+
+    /**
+     * set flag to run performance audits for page transitions
+     */
+    _enablePerformanceAudits ({ networkThrottling = DEFAULT_NETWORK_THROTTLING_STATE, cpuThrottling = 4, cacheEnabled = false } = {}) {
+        if (!NETWORK_STATES.hasOwnProperty(networkThrottling)) {
+            throw new Error(`Network throttling profile "${networkThrottling}" is unknown, choose between ${Object.keys(NETWORK_STATES).join(', ')}`)
+        }
+
+        if (typeof cpuThrottling !== 'number') {
+            throw new Error(`CPU throttling rate needs to be typeof number but was "${typeof cpuThrottling}"`)
+        }
+
+        this.networkThrottling = networkThrottling
+        this.cpuThrottling = cpuThrottling
+        this.cacheEnabled = Boolean(cacheEnabled)
+        this.shouldRunPerformanceAudits = true
+    }
+
+    /**
+     * custom command to disable performance audits
+     */
+    _disablePerformanceAudits () {
+        delete this.networkThrottling
```

Comment:
It might be better reset these parameters to some default values instead of wiping from `this`

Reply:
they are automatically defaulted when enabling performance audits .. I can remove the deletes though